Library/Homebrew/cli_parser.rb

```diff
@@ -225,7 +233,13 @@ def check_conflicts
 
           next if violations.count < 2
 
-          raise OptionConflictError, violations.map(&method(:name_to_option))
+          env_var_options = violations.select do |option|
+            @switch_sources[option_to_name(option)] == :env_var
```

Comment:
`:env_var` should be `:env`?

Reply:
Sounds right. Has this been fixed yet? I should also probably re-write the tests to catch this bug in the future

Reply:
Not yet!

Reply:
Yeah, the tests should ideally catch this failure

Reply:
Sounds good, will take care of this today

Reply:
So, looks like the issue with the test is that stubbing env vars isn't working at all. My tests unfortunately were structured in a way that they passed regardless of that. For example, this fails:

```ruby
allow(ENV).to receive(:[]).with("HOMEBREW_SWITCH_A").and_return("1")
expect(ENV["HOMEBREW_SWITCH_A"]).to eq("1") # actually is nil
```

Any idea how to fix this?

Reply:
Found the issue, I had my stubs in the wrong order. Pushing a change now, sorry for the bugs!